### ISSUE TITLE
Improve experience with DOI fetches

### ISSUE THREAD

**Issue by fwkoch** [executablebooks/mystmd#1335]
Changes to doi.org have caused headaches with resolving DOIs - particularly, rate-limiting 429 responses prevent parallel requests. Once we start getting 429s, we also see these as failures and do _more_ requests (handle.net, different content-type, etc). These just pile up, slow down the build, and never resolve.

So far, we've added limiting to outgoing requests at a per-page level. This helps with small projects, but doesn't apply to multiple page builds in parallel.

We need to:
- [x] remove requests to handle.net
- [x] limit outgoing requests at a site-build level, rather than page level
- [ ] stop unnecessary requests, e.g. based on a specific failure response from bibtex request, can we decide not to try csl json?

**Comment by fwkoch** [comment]
From my experience, handling the first two bullets here resolved the slow DOI fetches; the third is not necessary. These were released with v1.2.9

We can always re-open or create a new issue if more problems with DOIs arise...
